NewsDistribution/NewsServer.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net;
+using System.Net.Sockets;
+
+namespace NewsDistribution;
+
+public class NetworkClient
+{
+    public readonly Socket Socket;
+    public readonly string Name;
+
+    public NetworkClient(Socket socket, string name)
+    {
+        Socket = socket;
+        Name = name;
+    }
+}
+
+public class NewsServer
+{
+    private readonly Dictionary<string, NetworkClient> _clients = new();
+    private Socket? _listener;
+    private Thread? _acceptThread;
+    private CancellationTokenSource? _shouldAcceptClients;
+
+    public Dictionary<string, NetworkClient> Clients
```

Comment:
We can replace this with auto-property in one line ```public Dictionary<string, NetworkClient> Clients { get; } = new();```

Reply:
Yeah it was temporary, fixed now.